prime-dotnet-webapi/Models/Api/PaginatedList.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Prime.Models.Api
+{
+    public class PaginatedResponse<T>
+    {
+        public List<T> Results { get; set; }
+        public int Page { get; set; }
+        public int TotalPages { get; set; }
+        public int Total { get; set; }
+        public int PageSize { get; set; }
+
+        public PaginatedResponse(PaginatedList<T> list)
+        {
+            Results = list;
+            Page = list.Page;
+            TotalPages = list.TotalPages;
+            Total = list.Total;
+            PageSize = list.PageSize;
+        }
+    }
+    public class PaginatedList<T> : List<T>
+    {
+        private static int GetPageSize()
+        {
+            return 100;
+        }
+
+        public int Page { get; private set; }
+        public int TotalPages { get; private set; }
+        public int Total { get; private set; }
+        public int PageSize { get; private set; }
+
+        public PaginatedList(List<T> items, int count, int page, int pageSize)
+        {
+            Page = page;
+            TotalPages = (int)Math.Ceiling(count / (double)pageSize);
+            Total = count;
+            PageSize = pageSize;
+
+            AddRange(items);
+        }
+
+        public bool HasPreviousPage
+        {
+            get
+            {
+                return Page > 1;
+            }
+        }
+
+        public bool HasNextPage
+        {
+            get
+            {
+                return Page < TotalPages;
+            }
+        }
+
+        public PaginatedResponse<T> Response
+        {
+            get => new(this);
+        }
+
+        public static async Task<PaginatedList<T>> CreateAsync(IQueryable<T> source, int page)
+        {
+            var pageNumber = Math.Abs(page);
+            var count = await source.CountAsync();
+
+            pageNumber = (pageNumber * GetPageSize() < count) ? pageNumber : Convert.ToInt32(Math.Ceiling((double)count / GetPageSize()));
+
+            var items = await source.Skip((pageNumber - 1) * GetPageSize()).Take(GetPageSize()).ToListAsync();
```

Comment:
Getting this exception
```

[08:31:36 INF] HTTP POST /api/admins responded 200 in 767.3427 ms
[08:31:36 INF] Incoming request:  OPTIONS /api/enrollees
[08:31:36 INF] HTTP OPTIONS /api/enrollees responded 204 in 0.5975 ms
[08:31:36 INF] Incoming request:  GET /api/enrollees
[08:31:36 ERR] Failed executing DbCommand (15ms) [Parameters=[@__p_1='?' (DbType = Int32), @__p_0='?' (DbType = Int32)], CommandType='Text', CommandTimeout='30']
SELECT a3."IDIR", t."AlwaysManual", (
    SELECT e."StatusDate"
    FROM "EnrolmentStatus" AS e
    WHERE (t."Id" = e."EnrolleeId") AND (e."StatusCode" = 2)
    ORDER BY e."StatusDate" DESC, e."Id" DESC
    LIMIT 1), (
    SELECT e0."StatusDate"
    FROM "EnrolmentStatus" AS e0
    WHERE (t."Id" = e0."EnrolleeId") AND (e0."StatusCode" = 3)
    ORDER BY e0."StatusDate" DESC, e0."Id" DESC
    LIMIT 1), ((
    SELECT s."Confirmed"
    FROM "Submission" AS s
    WHERE t."Id" = s."EnrolleeId"
    ORDER BY s."CreatedDate" DESC
    LIMIT 1) = TRUE) AND ((
    SELECT s."Confirmed"
    FROM "Submission" AS s
    WHERE t."Id" = s."EnrolleeId"
    ORDER BY s."CreatedDate" DESC
    LIMIT 1) IS NOT NULL), (
    SELECT e1."StatusCode"
    FROM "EnrolmentStatus" AS e1
    WHERE t."Id" = e1."EnrolleeId"
    ORDER BY e1."StatusDate" DESC, e1."Id" DESC
    LIMIT 1), t."DateOfBirth", t."Id" + 1000, t."Email", (
    SELECT a."ExpiryDate"
    FROM "Agreement" AS a
    WHERE (t."Id" = a."EnrolleeId") AND (a."AcceptedDate" IS NOT NULL)
    ORDER BY a."CreatedDate" DESC
    LIMIT 1), t."FirstName", t."GivenNames", t."GPID", EXISTS (
    SELECT DISTINCT 1
    FROM "AgreementVersion" AS a0
    WHERE ((
        SELECT a1."Id"
        FROM "AgreementVersion" AS a1
        WHERE a1."AgreementType" = a0."AgreementType"
        ORDER BY a1."EffectiveDate" DESC
        LIMIT 1) = (
        SELECT a2."AgreementVersionId"
        FROM "Agreement" AS a2
        WHERE (t."Id" = a2."EnrolleeId") AND (a2."AcceptedDate" IS NOT NULL)
        ORDER BY a2."CreatedDate" DESC
        LIMIT 1)) OR (((
        SELECT a1."Id"
        FROM "AgreementVersion" AS a1
        WHERE a1."AgreementType" = a0."AgreementType"
        ORDER BY a1."EffectiveDate" DESC
        LIMIT 1) IS NULL) AND ((
        SELECT a2."AgreementVersionId"
        FROM "Agreement" AS a2
        WHERE (t."Id" = a2."EnrolleeId") AND (a2."AcceptedDate" IS NOT NULL)
        ORDER BY a2."CreatedDate" DESC
        LIMIT 1) IS NULL))), t."Id", t."LastName", COALESCE(CASE
    WHEN (e7."Id" IS NULL) THEN e8."EnrolleeId"
    ELSE e7."PaperEnrolleeId"
END, 0), EXISTS (
    SELECT 1
    FROM "Enrollee" AS e2
    WHERE (((e2."GPID" IS NOT NULL) AND (e2."GPID" LIKE 'NOBCSC%')) AND NOT (EXISTS (
        SELECT 1
        FROM "EnrolleeLinkedEnrolment" AS e3
        WHERE e3."PaperEnrolleeId" = e2."Id"))) AND (date_trunc('day', e2."DateOfBirth") = date_trunc('day', t."DateOfBirth"))), EXISTS (
    SELECT 1
    FROM "EnrolleeRemoteUser" AS e4
    WHERE t."Id" = e4."EnrolleeId"), NOT ((
    SELECT s0."Confirmed"
    FROM "Submission" AS s0
    WHERE t."Id" = s0."EnrolleeId"
    ORDER BY s0."CreatedDate" DESC
    LIMIT 1)) AND (((
    SELECT e5."StatusCode"
    FROM "EnrolmentStatus" AS e5
    WHERE t."Id" = e5."EnrolleeId"
    ORDER BY e5."StatusDate" DESC, e5."Id" DESC
    LIMIT 1 OFFSET 1) = 3) AND ((
    SELECT e5."StatusCode"
    FROM "EnrolmentStatus" AS e5
    WHERE t."Id" = e5."EnrolleeId"
    ORDER BY e5."StatusDate" DESC, e5."Id" DESC
    LIMIT 1 OFFSET 1) IS NOT NULL)), a3."Id", e7."Id", e8."Id", e9."CareSettingCode", e9."Id"
FROM (
    SELECT e6."Id", e6."AdjudicatorId", e6."AlwaysManual", e6."DateOfBirth", e6."Email", e6."FirstName", e6."GPID", e6."GivenNames", e6."LastName"
    FROM "Enrollee" AS e6
    ORDER BY e6."Id"
    LIMIT @__p_1 OFFSET @__p_0
) AS t
LEFT JOIN "Admin" AS a3 ON t."AdjudicatorId" = a3."Id"
LEFT JOIN "EnrolleeLinkedEnrolment" AS e7 ON t."Id" = e7."EnrolleeId"
LEFT JOIN "EnrolleeLinkedEnrolment" AS e8 ON t."Id" = e8."PaperEnrolleeId"
LEFT JOIN "EnrolleeCareSetting" AS e9 ON t."Id" = e9."EnrolleeId"
ORDER BY t."Id", a3."Id", e7."Id", e8."Id", e9."Id"
[08:31:36 ERR] An exception occurred while iterating over the results of a query for context type 'Prime.ApiDbContext'.
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.13\System.IO.MemoryMappedFiles.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
[08:31:38 ERR] HTTP GET /api/enrollees responded 500 in 1722.7940 ms
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Prime.Models.Api.PaginatedList`1.CreateAsync(IQueryable`1 source, Int32 page) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Models\Api\PaginatedList.cs:line 76
   at Prime.Services.EnrolleeService.GetEnrolleesAsync(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Services\EnrolleeService.cs:line 173
   at Prime.Controllers.EnrolleesController.GetEnrollees(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Controllers\EnrolleesController.cs:line 61
   at lambda_method1249(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Prime.Helpers.RequestLoggingMiddleware.Invoke(HttpContext context) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Helpers\RequestLoggingMiddleware.cs:line 33
   at Sentry.AspNetCore.SentryTracingMiddleware.InvokeAsync(HttpContext context) in /_/src/Sentry.AspNetCore/SentryTracingMiddleware.cs:line 106
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
[08:31:38 ERR] An unhandled exception has occurred while executing the request.
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Prime.Models.Api.PaginatedList`1.CreateAsync(IQueryable`1 source, Int32 page) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Models\Api\PaginatedList.cs:line 76
   at Prime.Services.EnrolleeService.GetEnrolleesAsync(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Services\EnrolleeService.cs:line 173
   at Prime.Controllers.EnrolleesController.GetEnrollees(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Controllers\EnrolleesController.cs:line 61
   at lambda_method1249(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Prime.Helpers.RequestLoggingMiddleware.Invoke(HttpContext context) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Helpers\RequestLoggingMiddleware.cs:line 33
   at Sentry.AspNetCore.SentryTracingMiddleware.InvokeAsync(HttpContext context) in /_/src/Sentry.AspNetCore/SentryTracingMiddleware.cs:line 106
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
Loaded 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\5.0.13\Microsoft.AspNetCore.Http.Extensions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
[08:31:38 INF] Incoming request:  OPTIONS /api/client-logs
[08:31:38 INF] HTTP OPTIONS /api/client-logs responded 204 in 0.8038 ms
[08:31:38 INF] Incoming request:  GET /api/enrollees
[08:31:38 INF] Incoming request:  POST /api/client-logs
[08:31:38 ERR] Failed executing DbCommand (15ms) [Parameters=[@__p_1='?' (DbType = Int32), @__p_0='?' (DbType = Int32)], CommandType='Text', CommandTimeout='30']
SELECT a3."IDIR", t."AlwaysManual", (
    SELECT e."StatusDate"
    FROM "EnrolmentStatus" AS e
    WHERE (t."Id" = e."EnrolleeId") AND (e."StatusCode" = 2)
    ORDER BY e."StatusDate" DESC, e."Id" DESC
    LIMIT 1), (
    SELECT e0."StatusDate"
    FROM "EnrolmentStatus" AS e0
    WHERE (t."Id" = e0."EnrolleeId") AND (e0."StatusCode" = 3)
    ORDER BY e0."StatusDate" DESC, e0."Id" DESC
    LIMIT 1), ((
    SELECT s."Confirmed"
    FROM "Submission" AS s
    WHERE t."Id" = s."EnrolleeId"
    ORDER BY s."CreatedDate" DESC
    LIMIT 1) = TRUE) AND ((
    SELECT s."Confirmed"
    FROM "Submission" AS s
    WHERE t."Id" = s."EnrolleeId"
    ORDER BY s."CreatedDate" DESC
    LIMIT 1) IS NOT NULL), (
    SELECT e1."StatusCode"
    FROM "EnrolmentStatus" AS e1
    WHERE t."Id" = e1."EnrolleeId"
    ORDER BY e1."StatusDate" DESC, e1."Id" DESC
    LIMIT 1), t."DateOfBirth", t."Id" + 1000, t."Email", (
    SELECT a."ExpiryDate"
    FROM "Agreement" AS a
    WHERE (t."Id" = a."EnrolleeId") AND (a."AcceptedDate" IS NOT NULL)
    ORDER BY a."CreatedDate" DESC
    LIMIT 1), t."FirstName", t."GivenNames", t."GPID", EXISTS (
    SELECT DISTINCT 1
    FROM "AgreementVersion" AS a0
    WHERE ((
        SELECT a1."Id"
        FROM "AgreementVersion" AS a1
        WHERE a1."AgreementType" = a0."AgreementType"
        ORDER BY a1."EffectiveDate" DESC
        LIMIT 1) = (
        SELECT a2."AgreementVersionId"
        FROM "Agreement" AS a2
        WHERE (t."Id" = a2."EnrolleeId") AND (a2."AcceptedDate" IS NOT NULL)
        ORDER BY a2."CreatedDate" DESC
        LIMIT 1)) OR (((
        SELECT a1."Id"
        FROM "AgreementVersion" AS a1
        WHERE a1."AgreementType" = a0."AgreementType"
        ORDER BY a1."EffectiveDate" DESC
        LIMIT 1) IS NULL) AND ((
        SELECT a2."AgreementVersionId"
        FROM "Agreement" AS a2
        WHERE (t."Id" = a2."EnrolleeId") AND (a2."AcceptedDate" IS NOT NULL)
        ORDER BY a2."CreatedDate" DESC
        LIMIT 1) IS NULL))), t."Id", t."LastName", COALESCE(CASE
    WHEN (e7."Id" IS NULL) THEN e8."EnrolleeId"
    ELSE e7."PaperEnrolleeId"
END, 0), EXISTS (
    SELECT 1
    FROM "Enrollee" AS e2
    WHERE (((e2."GPID" IS NOT NULL) AND (e2."GPID" LIKE 'NOBCSC%')) AND NOT (EXISTS (
        SELECT 1
        FROM "EnrolleeLinkedEnrolment" AS e3
        WHERE e3."PaperEnrolleeId" = e2."Id"))) AND (date_trunc('day', e2."DateOfBirth") = date_trunc('day', t."DateOfBirth"))), EXISTS (
    SELECT 1
    FROM "EnrolleeRemoteUser" AS e4
    WHERE t."Id" = e4."EnrolleeId"), NOT ((
    SELECT s0."Confirmed"
    FROM "Submission" AS s0
    WHERE t."Id" = s0."EnrolleeId"
    ORDER BY s0."CreatedDate" DESC
    LIMIT 1)) AND (((
    SELECT e5."StatusCode"
    FROM "EnrolmentStatus" AS e5
    WHERE t."Id" = e5."EnrolleeId"
    ORDER BY e5."StatusDate" DESC, e5."Id" DESC
    LIMIT 1 OFFSET 1) = 3) AND ((
    SELECT e5."StatusCode"
    FROM "EnrolmentStatus" AS e5
    WHERE t."Id" = e5."EnrolleeId"
    ORDER BY e5."StatusDate" DESC, e5."Id" DESC
    LIMIT 1 OFFSET 1) IS NOT NULL)), a3."Id", e7."Id", e8."Id", e9."CareSettingCode", e9."Id"
FROM (
    SELECT e6."Id", e6."AdjudicatorId", e6."AlwaysManual", e6."DateOfBirth", e6."Email", e6."FirstName", e6."GPID", e6."GivenNames", e6."LastName"
    FROM "Enrollee" AS e6
    ORDER BY e6."Id"
    LIMIT @__p_1 OFFSET @__p_0
) AS t
LEFT JOIN "Admin" AS a3 ON t."AdjudicatorId" = a3."Id"
LEFT JOIN "EnrolleeLinkedEnrolment" AS e7 ON t."Id" = e7."EnrolleeId"
LEFT JOIN "EnrolleeLinkedEnrolment" AS e8 ON t."Id" = e8."PaperEnrolleeId"
LEFT JOIN "EnrolleeCareSetting" AS e9 ON t."Id" = e9."EnrolleeId"
ORDER BY t."Id", a3."Id", e7."Id", e8."Id", e9."Id"
[08:31:38 ERR] An exception occurred while iterating over the results of a query for context type 'Prime.ApiDbContext'.
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
Exception thrown: 'Npgsql.PostgresException' in System.Private.CoreLib.dll
[08:31:39 ERR] HTTP GET /api/enrollees responded 500 in 1213.4546 ms
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Prime.Models.Api.PaginatedList`1.CreateAsync(IQueryable`1 source, Int32 page) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Models\Api\PaginatedList.cs:line 76
   at Prime.Services.EnrolleeService.GetEnrolleesAsync(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Services\EnrolleeService.cs:line 173
   at Prime.Controllers.EnrolleesController.GetEnrollees(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Controllers\EnrolleesController.cs:line 61
   at lambda_method1249(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Prime.Helpers.RequestLoggingMiddleware.Invoke(HttpContext context) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Helpers\RequestLoggingMiddleware.cs:line 33
   at Sentry.AspNetCore.SentryTracingMiddleware.InvokeAsync(HttpContext context) in /_/src/Sentry.AspNetCore/SentryTracingMiddleware.cs:line 106
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
[08:31:39 ERR] An unhandled exception has occurred while executing the request.
Npgsql.PostgresException (0x80004005): 2201X: OFFSET must not be negative
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at Prime.Models.Api.PaginatedList`1.CreateAsync(IQueryable`1 source, Int32 page) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Models\Api\PaginatedList.cs:line 76
   at Prime.Services.EnrolleeService.GetEnrolleesAsync(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Services\EnrolleeService.cs:line 173
   at Prime.Controllers.EnrolleesController.GetEnrollees(EnrolleeSearchOptions searchOptions) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Controllers\EnrolleesController.cs:line 61
   at lambda_method1249(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Prime.Helpers.RequestLoggingMiddleware.Invoke(HttpContext context) in C:\Users\228884\Desktop\Projects\Repos\moh-prime\prime-dotnet-webapi\Helpers\RequestLoggingMiddleware.cs:line 33
   at Sentry.AspNetCore.SentryTracingMiddleware.InvokeAsync(HttpContext context) in /_/src/Sentry.AspNetCore/SentryTracingMiddleware.cs:line 106
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
  Exception data:
    Severity: ERROR
    SqlState: 2201X
    MessageText: OFFSET must not be negative
    File: nodeLimit.c
    Line: 264
    Routine: recompute_limits
[08:31:39 INF] Incoming request:  POST /api/client-logs
```

I also keep seeing this query loading
![image](https://user-images.githubusercontent.com/34261366/156027777-cf6b4f9b-7acd-4990-a3bd-76e1f7c7c0cc.png)

Reply:
Yeah it looks like the corner cases messed up. Should be good now